pkg/kubelet/dockertools/docker.go

```diff
@@ -180,11 +180,27 @@ func matchImageTagOrSHA(inspected dockertypes.ImageInspect, image string) bool {
 	}
 
 	if isDigested {
-		algo := digest.Digest().Algorithm().String()
-		sha := digest.Digest().Hex()
-		// Look specifically for short and long sha(s)
-		if strings.Contains(inspected.ID, algo+":"+sha) {
-			// We found the short or long SHA requested
+		for _, repoDigest := range inspected.RepoDigests {
+			named, err := dockerref.ParseNamed(repoDigest)
+			if err != nil {
+				glog.V(4).Infof("couldn't parse image RepoDigest reference %q: %v", repoDigest, err)
+				continue
+			}
+			if d, isDigested := named.(dockerref.Digested); isDigested {
+				if digest.Digest().Algorithm().String() == d.Digest().Algorithm().String() &&
+					digest.Digest().Hex() == digest.Digest().Hex() {
```

Comment:
Just as an FYI, if we get into this for loop, it means:

1. `pod.containers[i].image` is digested
1. the image exists on the node
1. the kubelet was able to `docker inspect` the image using the digest

It should be impossible to get here and **not** have a match between `pod.containers[i].image` and the value from `RepoDigests`.

We should be defensive, so I'm not suggesting we remove or change anything here. Just an FYI